RestApi/Sepes.RestApi/Controllers/StudyController.cs

```diff
@@ -24,13 +27,70 @@ public StudyController(ISepesDb dbService, IStudyService studyService)
         }
 
         [HttpPost("save")]
-        public async Task<StudyInput> SaveStudy([FromBody] StudyInput[] studies)
+        public async Task<ActionResult<StudyInput>> SaveStudy([FromBody] StudyInput[] studies)
```

Comment:
Most of the code here should be inside StudyService.Save().

The controller pattern is:
Try: Happy path.
Series of catch for known exceptions to user that gets sent back as errors to front-end.

And anything that is not handled the framework will give a 500 error

Reply:
As said in earlier meeting I was unable to get C# to allow me to pass BadRequest, StatusCode or Ok wrappers back to the controller. So unfortunately the error handling ended up there.

Reply:
That is where custom Exception comes in. where you defined your own exception classes like. SepesResourceNotFoundExseption

Reply:
And its the controller job to covert from internal error to front end errors.

Reply:
Good to know